kroxylicious-filters/kroxylicious-record-encryption/src/main/java/io/kroxylicious/filter/encryption/RecordEncryptionFilter.java

```diff
@@ -57,6 +61,27 @@ public class RecordEncryptionFilter<K>
         this.encryptionManager = encryptionManager;
         this.decryptionManager = decryptionManager;
         this.filterThreadExecutor = filterThreadExecutor;
+        checkCipherSuite();
+    }
+
+    private static void checkCipherSuite() {
+        checkCipherSuite(CipherSpec::newCipher);
+    }
+
+    /* exposed for testing */ static void checkCipherSuite(Function<CipherSpec, Cipher> cipherFunc) {
+        List<CipherSpec> failures = Arrays.stream(CipherSpec.values()).flatMap(cipherSpec -> {
+            try {
+                cipherFunc.apply(cipherSpec);
+                return Stream.empty();
+            }
+            catch (Exception e) {
+                log.error("A Cipher could not be constructed for CipherSpec {}", cipherSpec, e);
+                return Stream.of(cipherSpec);
+            }
+        }).toList();
+        if (!failures.isEmpty()) {
+            throw new EncryptionConfigurationException("Cipher Suite check failed, one or more ciphers could not be loaded: " + failures);
+        }
```

Comment:
yeah I think the enum names are good for the exception, have changed it, but I stuck with testing a List for emptiness, not as keen on using string emptiness, but it's no strong preference.

Reply:
Yeah thats fair. I'd attempted to avoid if by using the exception text as a prefix to the joiner but then realised I didn't then get a sensible test for when to throw or not.